### DISCUSSION TITLE
Manually Assigning an ID to a field array

### DISCUSSION THREAD

**Discussion by raywong-fb** [react-hook-form/react-hook-form#6364] [Q&A] [answered]
Hi and thank you in advanced!

I recently noticed that if I try to `append` a new item to a field array and I manually assign that item a unique `id`, the `watched` values appears to strip away the `id` from all fields on the next render (see CSB example).

Interestingly, the form data itself still appears to store the correct `id` values for each field.

In the CSB, you'll notice the stringified array at the bottom with the correct `id`s. Once you press the Append button, the new item gets added but in the stringified output, all of the `id`s have been stripped.

Any help in understanding what's happening and if there's a potential workaround would be much appreciated.

https://codesandbox.io/s/react-hook-form-manual-id-for-field-arrays-jx4z9

**Comment by bluebill1049** [comment, marked as answer]
maybe it's better to assign `useFieldArray` with a different keyName?

**Reply by raywong-fb** [reply]
I see. Is there a reason why it's stripping out the id though?

**Reply by bluebill1049** [reply]
i think it's due to default name is `id` for keyName, which is why by default its getting striped away.
